internal/service/cloudbackupschedule/resource_cloud_backup_schedule.go

```diff
@@ -492,9 +492,9 @@ func cloudBackupScheduleCreateOrUpdate(ctx context.Context, connV2 *admin.APICli
 	}
 
 	req := &admin.DiskBackupSnapshotSchedule{}
-
-	if v, ok := d.GetOk("copy_settings"); ok && len(v.([]any)) > 0 {
-		req.CopySettings = expandCopySettings(v.([]any))
+	copySettings := d.Get("copy_settings")
+	if copySettings != nil && (conversion.HasElementsSliceOrMap(copySettings) || d.HasChange("copy_settings")) {
```

Comment:
i think that would work but I would probably go better for this to make ir more robust and avoid any possible panic:

if copy, ok := copySettings.([]any); ok && (len(copy) > 0  || d.HasChange("copy_settings") {
		req.CopySettings = expandCopySettings(copy)
}

Reply:
Thanks for all the feedback 🚀 . I cannot see how this can panic now (maybe if the settings are not nil and not a slice? 
I prefer keeping it in two lines; seems more readable to me.

Reply:
exactly:  if the settings are not nil and not a slice
maybe with the current logic it can't happen now, but imagine there are some code changes at some moment, at the end it's a best practice to err on the side of precaution and always check type castings are valid

Reply:
I can see both options used in the code. I'll stick to the safe side and check type castings in the future.
Thank you for the feedback and patience strongly appreciated ❤️

Reply:
@EspenAlbert very subtle but I think @lantoli  might have a point here. `HasElementsSliceOrMap` will return true if it's a map and the following line could fail. Am I correct? 
Now, that field shouldn't be a map but it's a safer move to explicitly check it's a slice.

Let me know if I am missing something

Reply:
@marcosuma yes you are correct.
But is it worth another PR? I can see multiple usages of unsafe type castings in the code:
- https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/advancedcluster/model_advanced_cluster.go#L405
- https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/advancedcluster/model_advanced_cluster.go#L509
- https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go#L324
- +++ (searching for `.([]any)`

I can do an aggregate PR to fix these.
WDYT @lantoli @marcosuma ?

Reply:
Update: Found out that we can add:
```yaml
linters-settings:
  errcheck:
    check-type-assertions: true
```
to our [golangci.yml file](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/.golangci.yml)

Some of the example errors:
```
internal/common/conversion/flatten_expand.go:32:10: Error return value is not checked (errcheck)
        list := d.Get("tags").(*schema.Set)
                ^
internal/common/conversion/flatten_expand.go:35:10: Error return value is not checked (errcheck)
                tag := item.(map[string]any)
                       ^
internal/common/conversion/flatten_expand.go:54:12: Error return value is not checked (errcheck)
                res[i] = v.(string)
```

Reply:
@EspenAlbert I really like the idea of a PR activating that linter and updating the code

Reply:
Create a ticket: CLOUDP-260150